meson.build

```diff
@@ -243,6 +243,20 @@ else
   conf.set('fallthrough', 'do {} while (0) /* fallthrough */')
 endif
 
+if cc.has_function('TEMP_FAILURE_RETRY', prefix : '#include <errno.h>')
+  conf.set('TFR', 'TEMP_FAILURE_RETRY')
+else
+  conf.set('TFR(exp)', '''					\
+({								\
+	long int __result = 0; 					\
+	do {							\
+		__result = (long int)(exp);			\
+	} while ((__result == -1) && (errno == EINTR));		\
```

Comment:
Could we be stuck here for a very long time?

`EINTR` happens when a signal is received. For example, if someone types CTRL-C (`SIGINT`) to stop a process that seems stuck. With this, CTRL-C will have no effect. Just wondering if that's what we want.

Reply:
Yes, that is a problem. Though I am not sure what /dev/nvme-fabrics does on SIGINT.

Reply:
> > Yes, this is a problem. Though I am not sure what /dev/nvme-fabrics does on SIGINT.
> 
> Right! I remember being stuck on `/dev/nvme-fabrics` before and there is no way to get out. The kernel just doesn't allow interrupting operations on that device. So, maybe my comment does not apply here.

Indeed, though I think it should be possible to handle SIGINTs, it's just not there yet.

Reply:
After a bit of refreshing my memory on signals, I don't think there is a real problem. The ctrl-c is translated to SIGINT which is then delivered to the process. The default handler for SIGINT is to abort the process. That means when the syscall returns the default handler gets executed (terminate process) and thus the loop will not be run anymore.

Reply:
Removed the TFR macro from the open call, as this call shouldn't be a blocking call.